orchestration/dagster_orchestration/hca_orchestration/tests/solids/load_hca/data_files/test_load_data_metadata_files.py

```diff
@@ -0,0 +1,74 @@
+from dagster import SolidExecutionResult, execute_solid
+from dagster_utils.contrib.data_repo.typing import JobId
+
+from hca_orchestration.pipelines.load_hca import test_mode
+from hca_orchestration.solids.load_hca.data_files.load_data_metadata_files import ingest_metadata_for_file_type, \
+    ingest_metadata, file_metadata_fanout
+from hca_orchestration.support.typing import HcaScratchDatasetName, MetadataType, MetadataTypeFanoutResult
+
+run_config = {
+                "resources": {
+                    "scratch_config": {
+                        "config": {
+                            "scratch_bucket_name": "bucket_name",
+                            "scratch_prefix_name": "prefix_name",
+                            "scratch_bq_project": "bq_project",
+                            "scratch_dataset_prefix": "dataset_prefix",
+                            "scratch_table_expiration_ms": 86400000
+                        }
+                    },
+                    "target_hca_dataset": {
+                        "config": {
+                            "dataset_name": "dataset_name",
+                            "dataset_id": "dataset_id",
+                            "project_id": "project_id",
+                            "billing_profile_id": "billing_profile_id"
+                        }
+                    }
+                }
+            }
+metadata_fanout_result = MetadataTypeFanoutResult(
+    scratch_dataset_name=HcaScratchDatasetName("dataset"),
+    metadata_type=MetadataType("metadata"),
+    path="path"
+)
+
+
+def test_ingest_metadata_for_file_type():
+    result: SolidExecutionResult = execute_solid(
+        ingest_metadata_for_file_type,
+        mode_def=test_mode,
+        input_values={
+            "file_metadata_fanout_result": metadata_fanout_result
+        },
+        run_config=run_config
+    )
+
+    assert result.success
```

Comment:
🔧 I think there are stronger assertions to be made here. Given that this is a fairly side-effect heavy solid, what if instead of the `test_mode` from `load_hca` we define a custom `ModeDefinition` for the purpose of this test, and pass either a `ResourceDefinition.hardcoded_resource(<the mock>)` or just use `ResourceDefinition.mock_resource()`.

We something like that [here](https://github.com/DataBiosphere/hca-ingest/blob/f6703e4f48dc3a13642d1ffe1a48ba4a14e3da13/orchestration/dagster_orchestration/hca_orchestration/tests/solids/test_load_data_files.py#L23-L41) (with admittedly weak assertions).

Reply:
Ah, I see you explicitly mentioned punting on the stronger assertions as part of this PR, I'll buy that. I think it'd still make sense to introduce the mocks to pave the way for better assertions going forward.